android/src/main/java/com/revenuecat/purchases_flutter/PurchasesFlutterPlugin.java

```diff
@@ -154,17 +154,22 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
                 type = call.argument("type");
                 Map<String, Object> presentedOfferingContext = call.argument("presentedOfferingContext");
                 List<Map<String, Object>> addOnStoreProducts = call.argument("addOnStoreProducts");
+                List<Map<String, Object>> addOnSubscriptionOptions = call.argument("addOnSubscriptionOptions");
                 purchaseProduct(productIdentifier, type, googleOldProductIdentifer, googleProrationMode,
-                        googleIsPersonalizedPrice, presentedOfferingContext, addOnStoreProducts, result);
+                        googleIsPersonalizedPrice, presentedOfferingContext, addOnStoreProducts,
+                        addOnSubscriptionOptions, result);
                 break;
             case "purchasePackage":
                 String packageIdentifier = call.argument("packageIdentifier");
                 presentedOfferingContext = call.argument("presentedOfferingContext");
                 googleOldProductIdentifer = call.argument("googleOldProductIdentifier");
                 googleProrationMode = call.argument("googleProrationMode");
                 googleIsPersonalizedPrice = call.argument("googleIsPersonalizedPrice");
+                addOnStoreProducts = call.argument("addOnStoreProducts");
```

Comment:
I'm not a fan of the pattern we're using here where the variables are declared in an earlier case in this switch...I might make a follow up PR to move away from this pattern 😅